types/emoji-mart/dist-es/utils/shared-props.d.ts

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+
+import { Data } from './data';
+import { EmojiData, EmojiSkin, CustomEmoji } from './emoji-index/nimble-emoji-index';
+
+export type BackgroundImageFn = (set: EmojiSet, sheetSize: EmojiSheetSize) => string;
+export type EmojiSet = 'apple' | 'google' | 'twitter' | 'emojione' | 'messenger' | 'facebook';
+export type EmojiSheetSize = 16 | 20 | 32 | 64;
+
+export interface EmojiProps {
+    onOver?(emoji: EmojiData, e: React.MouseEvent<HTMLElement>): void;
+    onLeave?(emoji: EmojiData, e: React.MouseEvent<HTMLElement>): void;
+    onClick?(emoji: EmojiData, e: React.MouseEvent<HTMLElement>): void;
+    fallback?(emoji: EmojiData, props: EmojiProps): React.Component;
+    /** defaults to returning a png from unpkg.com-hosted emoji-datasource-${set} */
+    backgroundImageFn?: BackgroundImageFn;
+    native?: boolean;
+    forceSize?: boolean;
+    tooltip?: boolean;
+    /** defaults to 1 */
+    skin?: EmojiSkin;
+    /** defaults to 64 */
+    sheetSize?: EmojiSheetSize;
+    /** defaults to 52 */
+    sheetColumns?: number;
+    /** defaults to 52 */
+    sheetRows?: number;
+    /** defaults to 'apple' */
+    set?: EmojiSet;
+    size: number;
+    emoji: string | EmojiData;
+    html?: boolean;
+}
```

Comment:
[there's a `data` prop here that's listed as required](https://github.com/missive/emoji-mart/blob/v2.8.0/src/utils/shared-props.js#L4) but [it's in the default props](https://github.com/missive/emoji-mart/blob/v2.8.0/src/components/emoji/emoji.js#L19), which makes it an optional prop.

Reply:
However, the docs tell you to use `NimbleEmoji` instead if you want to pass custom data, so I guess it's best to omit it here; maybe with a comment.

Reply:
Yeah I initially had data here, but then when testing had errors for obvious reasons. Will add comment.